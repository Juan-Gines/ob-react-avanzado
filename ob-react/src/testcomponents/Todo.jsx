import React from 'react';

const Todo = ({ todo }) => {
	const { id, text, completed } = todo;
	const [checked, setChecked] = React.useState(completed);
	return (
		<div data-testid={`todo-test-${id}`}>
			<label>
				<input
					id='checkbox'
					type='checkbox'
					checked={completed}
					onChange={() => setChecked(!checked)}
				/>
				{text}
			</label>
		</div>
	);
};

export default Todo;
