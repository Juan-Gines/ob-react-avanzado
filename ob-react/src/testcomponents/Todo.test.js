import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Todo from './Todo';

afterEach(() => cleanup());

describe('Todo component is working correctly', () => {
	const todo = { id: 5, text: 'Hacer la compra', completed: true };
	render(<Todo todo={todo} />);
	const todoelement = screen.getByTestId(`todo-test-${todo.id}`);

	it('should render Todo component', () => {
		expect(todoelement).toBeInTheDocument();
	});

	it('should have right textw', () => {
		expect(todoelement).toHaveTextContent(todo.text);
	});
});

describe('Todo component should check/uncheck completed/not completed todos', () => {
	const todo = { id: 12, text: 'Hacer la compra', completed: true };
	render(<Todo todo={todo} />);
	const todoelement = screen.getByTestId(`todo-test-${todo.id}`);
	const checkbox = todoelement.querySelector('#checkbox');

	it('should be checked', () => {
		expect(checkbox).toBeChecked();
	});

	it('matches snapshot', () => {
		const snapTodo = { id: 1, text: 'Pasar los tests 2', completed: false };
		const tree = renderer.create(<Todo todo={snapTodo} />);
		expect(tree).toMatchSnapshot();
	});
});
