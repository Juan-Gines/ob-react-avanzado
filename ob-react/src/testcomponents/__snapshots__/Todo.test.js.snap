// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component should check/uncheck completed/not completed todos matches snapshot 1`] = `
<div
  data-testid="todo-test-1"
>
  <label>
    <input
      checked={false}
      id="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    Pasar los tests 2
  </label>
</div>
`;
